source/material_model/rheology/strain_dependent.cc

```diff
@@ -360,6 +396,32 @@ namespace aspect
 
       }
 
+      template <int dim>
+      double
+      StrainDependent<dim>::
+      calculate_strain_healing(const MaterialModel::MaterialModelInputs<dim> &in,
+                               const double e_ii,
+                               const unsigned int j) const
+      {
+        const double reference_temperature = this->get_adiabatic_surface_temperature();
+        double healed_strain = e_ii;
+
+        switch (healing_mechanism)
+          {
+            case no_healing:
+            {
+              break;
+            }
+            case temperature_dependent:
+            {
+              healed_strain -= in.composition[j][0] * strain_healing_recovery_timescale *
+                               std::exp(-strain_healing_viscosity_prefactor * 0.5 * (1.0 - in.temperature[j]/reference_temperature))
+                               * this->get_timestep();
```

Comment:
Are you getting unrealistic changes in strain when `in.temperature[j]` >> `reference_temperature`?